File: backend/schemas/overview.py
"""
backend/schemas/overview.py
---------------------------
Pydantic models for the Home / Ecosystem Overview dashboard.
"""

from __future__ import annotations

from pydantic import BaseModel, Field
from backend.schemas.apps import AppListItem
from backend.schemas.categories import CategoryListItem


class StatCard(BaseModel):
    label: str
    value: int | float
    formatted: str
    description: str | None = None


class ReviewAvailability(BaseModel):
    total_active_apps: int
    apps_with_public_reviews: int
    apps_with_no_public_reviews: int
    total_stored_reviews: int
    apps_with_stored_reviews: int
    uncollected_apps_count: int


class OverviewDashboardResponse(BaseModel):
    summary: dict[str, StatCard]
    review_availability: ReviewAvailability
    pricing_distribution: dict[str, int]
    rating_distribution: dict[int, int]
    top_categories: list[CategoryListItem]
    top_reviewed_apps: list[AppListItem]
    top_rated_apps: list[AppListItem]

File: frontend/src/api/overview.ts
/**
 * frontend/src/api/overview.ts
 * API calls for Home / Ecosystem Overview.
 */

import { apiClient } from './client';
import type { AppItem, CategoryItem } from '../types';

export interface StatCardData {
  label: string;
  value: number | string;
  formatted: string;
  description?: string;
}

export interface ReviewAvailability {
  total_active_apps: number;
  apps_with_public_reviews: number;
  apps_with_no_public_reviews: number;
  total_stored_reviews: number;
  apps_with_stored_reviews: number;
  uncollected_apps_count: number;
}

export interface OverviewResponse {
  summary: {
    total_apps: StatCardData;
    total_categories: StatCardData;
    total_pricing_plans: StatCardData;
    total_reviews: StatCardData;
    average_rating: StatCardData;
  };
  review_availability: ReviewAvailability;
  pricing_distribution: Record<string, number>;
  rating_distribution: Record<string, number>;
  top_categories: CategoryItem[];
  top_reviewed_apps: AppItem[];
  top_rated_apps: AppItem[];
}

export async function fetchOverview(): Promise<OverviewResponse> {
  return apiClient<OverviewResponse>('/overview');
}
